sdk/anomalydetector/ai-anomaly-detector/package.json

```
@@ -65,13 +65,13 @@
     "@azure/core-http": "^1.2.0",
     "@azure/core-auth": "^1.2.0",
     "@azure/logger": "^1.0.0",
-    "@opentelemetry/api": "^0.10.2",
+    "@opentelemetry/api": "^1.0.0-rc.0",
     "tslib": "^2.0.0",
-    "@azure/core-tracing": "1.0.0-preview.10"
+    "@azure/core-tracing": "1.0.0-preview.11"
   },
   "devDependencies": {
     "@azure/dev-tool": "^1.0.0",
-    "@azure/identity": "^1.1.0",
+    "@azure/identity": "^2.0.0-beta.1",
```

I guess we have to do this for some testing? What happens if someone's app in the wild uses identity v1 with this though? We need to still be compatible, right?

---

For runtime dependencies, pre-release versions should be pinned instead of floated, since breaking changes are allowed across pre-release versions.  The same might be true for dev dependencies.  If so:

```suggestion
    "@azure/identity": "2.0.0-beta.1",
```

---

@xirzec : That's a good point. The `beta` changeover happened while I was crafting up this PR so I'm not entirely sure of the compatiblity plans. @sadasant - can you give me an idea of what I should do here? I don't know if your plan was to make 2.0 a min-bar in the future or if we were going to support 1.x and 2.x concurrently for awhile.

@mikeharder : yup, good point. Let me clarify what's happening with 2.0 with with @sadasant .

---

@richardpark-msft: Are you planning to pin the dependency version for identity?

---

I have a chat with @sadasant about it. I might need to actually rollback to 1.x and figure out the proper rollup incantation to support it. Or I will come up with a proper justification for this. :)

---

(We're still talking!)

Since this is a devDependency, it would be very useful to me if we could start using Identity ^2.0.0-beta.1 because I want to be able to have as many eyes (and automation) as possible to cover before we do the full GA. 🙏 what do you all think?

---

@xirzec , @mikeharder  - given our desire to move forward to 2.0 (and the idea that future betas are not API breaking) are we okay with this change?

---

A couple thoughts:

1. If we want to update `devDependencies` from `identity@1` to `identity@2`, should this be a separate PR?  It seems  unrelated to `opentelemetry`.

2. Even for `devDependencies`, I think pre-release versions should be pinned to `identity@2.0.0-beta.1`, rather than floated `identity@^2.0.0-beta.1`.  If there are breaking changes in `identity@2.0.0-beta.2`, we don't want our builds or tests to break.  Instead, upgrading from `identity@2.0.0-beta.1` from `identity@2.0.0-beta.2` should be a PR that we validate before merging.

---

@richardpark-msft: I stand by my recommendations above (to change identity in a separate PR, and pin the version instead of floating).  However, if the JS team wants to include the identity upgrade in this PR, and float the version, then I am OK with this as long as you understand the consequences that things might break in the future.

---

I am OK with floating pre-release only for `devDependencies`.  For prod `dependencies` I am still strongly against floating pre-release since it can break customers.

---

@mikeharder - I think for `dependences` you and I are in 100% agreement. I think this is a potential one-off, but since we want to GA at some point in the next couple of months we might as well get the testing in now.